### DISCUSSION TITLE
Is it possible to configure VSCode to run `forge fmt` on save?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [foundry-rs/foundry#4014] [General]
The Foundry Book documents the `forge fmt` command [here](https://book.getfoundry.sh/reference/config/formatter#int_types), but it doesn't give any indication for how to configure it to automatically format on Save in VSCode.

What would one typically do to enable format on save in VSCode is add the following settings:

```json
{
  "editor.formatOnSave": true,
  "[solidity]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode" 
  }
```

The trouble is that the `defaultFormatter` setting expects the name of an extension, but there isn't any bespoke Foundry VSCode extension at the moment. So it seems to be that it is not currently possible to format on save with `forge fmt` in VSCode - but I opened this discussion to make sure that my understanding is correct?

I am not sure if it possible to go around this limitation with [tasks](https://code.visualstudio.com/docs/editor/tasks). It's definitely possible to add a task for running `forge fmt`, but that wouldn't help much because the user would have to remember to run the task by themselves.

**Comment by PaulRBerg** [comment]
It looks like Juan Blanco's [vscode-solidity](https://github.com/juanfranblanco/vscode-solidity) extension has support for `forge fmt`. @shekhirin added support for Forge in [this PR](https://github.com/juanfranblanco/vscode-solidity/pull/359), and the configuration is as simple as adding this to your VSCode settings:

> "solidity.formatter": "forge"

This is not documented anywhere, however. I will a PR in the Foundry Book repo to document the set-up.

**Reply by mds1** [reply]
You can also do this with the [hardhat VSCode](https://github.com/NomicFoundation/hardhat-vscode) extension, which personally I prefer as it seems to work better in pure foundry projects

**Reply by PaulRBerg** [reply]
Oh, nice!

**Reply by just2102** [reply]
<img width="1320" height="212" alt="image" src="https://github.com/user-attachments/assets/78cfc364-dcd0-4b90-a1ef-3c704636e2be" />
I'm not sure this works.
<img width="1110" height="780" alt="image" src="https://github.com/user-attachments/assets/91d34af6-9ca7-4fbc-a65b-ceead42e0141" />

**Reply by mmsaki** [reply]
I have build auto formatting with forge fmt for neovim. 

I could build a similar extension on vscode but I don't use it much. @just2102 Let me know if this is still an issue, I could build the same thing for vscode. 

https://x.com/msakiart/status/1950325313894064610. I think foundry tooling needs more integrations on text editors e.g vscode nvim etc.
